packages/ui-file-drop/src/FileDrop/index.tsx

```diff
@@ -102,10 +103,13 @@ class FileDrop extends Component<FileDropProps, FileDropState> {
   constructor(props: FileDropProps) {
     super(props)
     // @ts-expect-error ts-migrate(2322) FIXME: Type 'string' is not assignable to type 'null'.
-    this.defaultId = uid('FileDrop')
-    ;(this as any).messagesId = uid('FileDrop-messages')
+    this.defaultId = props.deterministicId!
+    this.messagesId = generateId(
+      'FileDrop-messages',
+      //@ts-expect-error props.instanceMapCounter
+      props.instanceMapCounter
+    )
```

Comment:
I'm a bit confused about how this util is used now. I some components you use `generateId` and some other places the `this.props.deterministicId` and here it is both :) I think I get it, the `deterministicId` is supposed to give back the id with the componentId and you can generate extra id-s with the other util? 

How about making this prop accept a function too? So by default `props.deterministicId =>  'FileDrop_1'`, but it would also accept `props.deterministicId('FileDrop-messages') =>  'FileDrop-messages_1'`? If you have a better idea, that is fine too, just would be nice to clear the confusion and the 2 different usages :)

Reply:
The idea was that the default behaviour would be passed down as a simple prop (`deterministicId`) when the component would just use essentially it's componentId for id generation.
For every other cases it would have control over how to generate id.

But I agree this is not the most elegant way of doing it.
Sadly I don't see how you suggestion might work, how to pass down a string in 1 case, and a function in another?
What about `generateId` would passed down as a function and it would accept an optional parameter - which would be defaulted to `instanceName`=`component.componentId` - and it would generate based on this parameter?

Reply:
Sure that sounds good too!

Reply:
Great, should be fixed then!